xdsl/dialects/accfg.py

```diff
@@ -303,16 +281,180 @@ def verify_(self) -> None:
+            return name, val
+
+        args: Sequence[tuple[str, SSAValue]] = parser.parse_comma_separated_list(
+            Parser.Delimiter.PAREN, parse_itm
+        )
+
+        attributes = {}
+        if parser.parse_optional_keyword("attrs"):
+            attributes = parser.parse_optional_attr_dict()
+
+        parser.parse_punctuation(":")
+        res_typ = parser.parse_type()
+        setup_op = cls(
+            [val for _, val in args],
+            [name for name, _ in args],
+            accelerator,
+            in_state,
+        )
+        setup_op.out_state.type = res_typ
+        setup_op.attributes.update(attributes)
+        return setup_op
+
+
+class AcceleratorSymbolOpTrait(SymbolOpInterface):
+    def get_sym_attr_name(self, op: Operation) -> StringAttr | None:
+        assert isinstance(op, AcceleratorOp)
+        return StringAttr(op.name_prop.string_value())
+
+
+@irdl_op_definition
+class AcceleratorOp(IRDLOperation):
+    """
+    Declares an accelerator that can be configured, launched, etc.
+    `fields` is a dictionary mapping accelerator configuration names to
+    CSR addresses.
+    """
+
+    name = "accfg.accelerator"
+
+    traits = traits_def(AcceleratorSymbolOpTrait())
+
+    name_prop = prop_def(SymbolRefAttr, prop_name="name")
+
+    fields = prop_def(DictionaryAttr)
+
+    launch_fields = prop_def(DictionaryAttr)
+
+    barrier = prop_def(IntegerAttr[IntegerType])  # TODO: this will be reworked in a later version
+
+    def __init__(
+        self,
+        name: str | StringAttr | SymbolRefAttr,
+        fields: dict[str, int] | DictionaryAttr,
+        launch_fields: dict[str, int] | DictionaryAttr,
+        barrier: int | IntegerAttr[IntegerType],
+    ):
+        if not isinstance(fields, DictionaryAttr):
+            fields = DictionaryAttr(
+                {name: IntegerAttr(val, i32) for name, val in fields.items()}
+            )
+
+        if not isinstance(launch_fields, DictionaryAttr):
+            launch_fields = DictionaryAttr(
+                {name: IntegerAttr(val, i32) for name, val in launch_fields.items()}
+            )
+
+        super().__init__(
+            properties={
+                "name": (
+                    SymbolRefAttr(name) if not isinstance(name, SymbolRefAttr) else name
+                ),
+                "fields": fields,
+                "launch_fields": launch_fields,
+                "barrier": (
+                    IntegerAttr(barrier, i32)
+                    if not isinstance(barrier, IntegerAttr)
+                    else barrier
+                ),
+            }
+        )
+
+    def verify_(self) -> None:
+        for _, val in self.fields.data.items():
+            if not isinstance(val, IntegerAttr):
+                raise VerifyException("fields must only contain IntegerAttr!")
+
+    def field_names(self) -> tuple[str, ...]:
+        return tuple(self.fields.data.keys())
+
+    def field_items(self) -> Iterable[tuple[str, IntegerAttr]]:
```

Comment:
```suggestion
    def field_items(self) -> Iterable[tuple[str, IntegerAttr[Attribute]]]:
```

Reply:
Doesn't work :(

Do you know what's the best way here?

Reply:
Ah, found a fix. Apparently `AnyIntegerAttr` just works